map/src/main/kotlin/edu/artic/map/MapObjectDetailsFragment.kt

```diff
@@ -0,0 +1,131 @@
+package edu.artic.map
+
+import android.os.Bundle
+import android.view.View
+import com.bumptech.glide.Glide
+import com.fuzz.rx.bindTo
+import com.fuzz.rx.bindToMain
+import com.fuzz.rx.defaultThrottle
+import com.fuzz.rx.disposedBy
+import com.jakewharton.rxbinding2.view.clicks
+import com.jakewharton.rxbinding2.widget.text
+import edu.artic.analytics.ScreenCategoryName
+import edu.artic.audioui.BottomAudioPlayerFragment
+import edu.artic.db.models.ArticObject
+import edu.artic.media.audio.AudioPlayerService
+import edu.artic.viewmodel.BaseViewModelFragment
+import kotlinx.android.synthetic.main.fragment_map_object_details.*
+import kotlin.reflect.KClass
+
+/**
+ * @author Sameer Dhakal (Fuzz)
+ */
+class MapObjectDetailsFragment : BaseViewModelFragment<MapObjectDetailsViewModel>() {
+
+    override val viewModelClass: KClass<MapObjectDetailsViewModel>
+        get() = MapObjectDetailsViewModel::class
+
+    override val title: String
+        get() = ""
+
+    override val layoutResId: Int
+        get() = R.layout.fragment_map_object_details
+
+
+    override val screenCategory: ScreenCategoryName
+        get() = ScreenCategoryName.OnViewDetails
+
+    private val mapObject: ArticObject? by lazy { arguments?.getParcelable<ArticObject>(ARG_MAP_OBJECT) }
```

Comment:
since we dont allow it to be null when creating the fragment, we can safely assert here that its not null:
```
private val mapObject: ArticObject by lazy { arguments!!.getParcelable<ArticObject>(ARG_MAP_OBJECT) }
```

Reply:
or if you create `requireArguments()` then:
```
private val mapObject: ArticObject by lazy { requireArguments().getParcelable<ArticObject>(ARG_MAP_OBJECT) }
```